### ISSUE TITLE
feat: Support Filtering by multiple or filters

### ISSUE THREAD

**Issue by groobyming** [GreptimeTeam/promql-parser#83]
@yuanbohan Hi Yuanbo, I want to support or filter but even if I add something in promql.y I get the error "invalid promql query", I am new to yacc and I have used antlr before (anltr is relatively It's relatively simple, just modify the grammar definition, but yacc doesn't seem to work), so I hope you can give me some suggestions.

【victoriametrics or expression】
<img width="1100" alt="image" src="https://github.com/GreptimeTeam/promql-parser/assets/34335395/0d267107-f982-4e7f-8afe-c8c2c310c016">
https://docs.victoriametrics.com/keyconcepts/#filtering-by-multiple-or-filters

【promql.y】

```
label_match_list -> Result<Matchers, String>:
                label_match_list COMMA label_matcher { Ok($1?.append($3?)) }
        |       label_match_list LOR label_matcher { Ok($1?.append($3?)) }  // support or filter
        |       label_matcher { Ok(Matchers::empty().append($1?)) }
;

```

【error】
expression：
`a{label1="1" or label2="2"}`

```
 Err("invalid promql query")
```

**Comment by yuanbohan** [comment]
MetricsQL is an enhancement option, and our team has discussed it. But we have no plan to totally implement the extension, and have no milestone for it.

Thanks for your PR, and I will merge it if it works fine, as well as the test cases, and test coverage.

**Comment by groobyming** [comment]
@yuanbohan 
Thanks! Looking forward to the progress

**Comment by groobyming** [comment]
I have one more small question. Why is it still not supported to use or connect lable_macher after making the following modifications?
before:
```
label_match_list -> Result<Matchers, String>:
                label_match_list COMMA label_matcher { Ok($1?.append($3?)) }
        |       label_matcher { Ok(Matchers::empty().append($1?)) }
;
```

after:
```
label_match_list -> Result<Matchers, String>:
                label_match_list COMMA label_matcher { Ok($1?.append($3?)) }
        |       label_match_list LOR label_matcher { Ok($1?.append($3?)) }
        |       label_matcher { Ok(Matchers::empty().append($1?)) }
;
```

cargo build success!

**Comment by yuanbohan** [comment]
I have not tested yet, but I think it should be placed in `label_matchers` production

```
label_matchers -> Result<Matchers, String>:
                LEFT_BRACE label_match_list LOR label_match_list RIGHT_BRACE { $2 }
        |       LEFT_BRACE label_match_list RIGHT_BRACE { $2 }
        |       LEFT_BRACE label_match_list COMMA RIGHT_BRACE { $2 }
        |       LEFT_BRACE RIGHT_BRACE { Ok(Matchers::empty()) }
        |       LEFT_BRACE COMMA RIGHT_BRACE
                { Err("unexpected ',' in label matching, expected identifier or right_brace".into()) }
;
```

and maybe there SHOULD BE some special logic in [inside_braces ](https://github.com/GreptimeTeam/promql-parser/blob/main/src/parser/lex.rs#L563) for the `LOR`, maybe it is tokenized as identifier instead of the LOR token

**Comment by yuanbohan** [comment]
@groobyming If you're trying this way as I mentioned above, you should notice that it's not exactly tested.

```
LEFT_BRACE label_match_list LOR label_match_list RIGHT_BRACE { $2 }
```

NOTE the tail part, the `{ $2 }` is wrongly typed, and you MUST change it

**Comment by yuanbohan** [comment]
fixed by #84
